### ISSUE TITLE
Migration away from selenium-wire to "nodriver"

### ISSUE THREAD

**Issue by Titaniumtown** [klept0/MS-Rewards-Farmer#161]
Selenium-wire has been archived:
https://github.com/wkeeling/selenium-wire
> Selenium Wire is no longer being maintained. Thank you for your support and all your contributions.

There seems to be a complete replacement for selenium + selenium-wire + undetected-chromedriver:
https://github.com/ultrafunkamsterdam/nodriver

**Comment by klept0** [comment]
Ha this is what he had been working on and why UD Chromedriver was being neglected. I would like to get this tested and swap over so it alleviates some of the crashing errors we are getting with UD driver. I assigned us and if we would like to get some form of chat going feel free to suggest - I have TG/Discord etc

**Comment by Titaniumtown** [comment]
still relevant

**Comment by ttraxxrepo** [comment]
@klept0 id highly suggest checking out selenium base as well while you’re considering migration. 

There are a lot of quality of life features baked in that make developing selenium apps much more pleasant. as an example, every command (find, click, etc) automatically has the web driver wait built in so you don’t have to write it every time.

https://seleniumbase.io/help_docs/uc_mode/

**Comment by cal4** [comment]
Yeah, I've attempted to get rid of https://github.com/ultrafunkamsterdam/undetected-chromedriver if for anything else it seems like pretty crappy code (see how quit and close doesn't work).

Granted this dependency is inherited from https://github.com/charlesbel/Microsoft-Rewards-Farmer/blob/master/requirements.txt, it would be nice exploring if we can just get rid of it altogether. Or at the least come up with a use case where vanilla Selenium doesn't work (or causes trouble).

**Comment by klept0** [comment]
100% for a migration away from it - have not had the time for a dedicated period to sit and see where the specifics from the UD-CD are to remove/replace and get this integrated.

**Comment by HaveSomeRubicon** [comment]
Hi is there anyway I can help with migrating to nodriver? I've never used nodriver but I think I have the time to learn it and make a contribution in someway or another.

**Comment by cal4** [comment]
@HaveSomeRubicon think the plan is to move away from nodriver/undetected chromedriver/selenium wire and to just using plain ole vanilla selenium (until proven otherwise). See https://github.com/klept0/MS-Rewards-Farmer/issues/161#issuecomment-2301665475.

If you'd like to contribute though, feel to fork try removing these dependencies, or feel free to pick up any of the other enhancements.

**Comment by Titaniumtown** [comment]
not stale :p

**Comment by klept0** [comment]
yeah Im fixing that - ignore the message

**Comment by MeetBhingradiya** [comment]
use may Use Customised Ai for Faster Help with Custom Instructions of ChatGPT becouse Now they Public the Future "Searching in Web"
![image](https://github.com/user-attachments/assets/ce672944-286d-4b7d-ab44-918e153d04e4)

also i have some paid bot that i purchsed from unkonwn discord servers for almost 1.5$ now that no work more properly i went to share their config you can look this below config @klept0 
i used this bot more than 5 months but only problem in login & Same Query Problem due to Google Trends API
```json
{
    "browserSettings": {
        "browserName": "chrome",
        "mobileUserAgent": [
            "Mozilla/5.0 (Linux; Android 12; SM-N9750) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Mobile Safari/537.36"
        ],
        "desktopUserAgent": [
            "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
        ],
        "imagesDisabled": true,
        "imagesDisabledInSetupMode": false
    },
    "searchSettings": {
        "searchDelay": 10,
        "searchPoints": 3,
        "cooldownSearchCount": 4,
        "cooldownSearchDelay": 16,
        "searchMethod": "URL"
    },
    "botSettings": {
        "activities": true,
        "punchCardsOnly": false,
        "startTime": null,
        "onFinishAction": [],
        "setupMode": false,
        "runDaily": false,
        "threadsCount": 1,
        "setupAccountProfilesBeforeRun": false,
        "cooldownModeByDefault": false
    },
    "discordAlerts": {
        "isEnabled": true,
        "webhookUrl": "<Webhook>",
        "alertFormat": "embed"
    },
    "ipChanging": {
        "isEnabled": false,
        "airplaneModeToggling": false,
        "enabledInSetupMode": false
    },
    "LogLevel": 3
}
```

**Comment by Titaniumtown** [comment]
not stale
